Project/Source/Panels/PanelScene.cpp

```diff
@@ -248,6 +249,10 @@ void PanelScene::Update() {
 						transform->SetScale(scale);
 						break;
 					}
+
+					if ((selectedGameObject->GetMask().bitMask & static_cast<int>(MaskType::CAST_SHADOWS)) != 0) {
```

Comment:
I think it's not necessary since you are invalidating in ComponentTransform SetPosition, SetRotation, and SetScale. Skip it if I'm wrong